toolboxes/scripts/GRGTools.py

```diff
@@ -498,6 +498,103 @@ def execute(self, parameters, messages):
         out_grid = RG.Build(parameters[3].value)
         return out_grid
 
+class NumberFeatures(object):
+    '''
+    Number input features within a specified area.
+    '''
+    def __init__(self):
+        '''
+        Number Features constructor
+        '''
+        self.label = "Number Features"
+        self.description = "Number input point features within a selected area."
+        self.category = "Gridded Reference Graphic"
+
+    def getParameterInfo(self):
+        '''
+        Define parameter definitions
+        '''
+        input_area_features = arcpy.Parameter(name='input_area_features',
+                                              displayName='Input Area to Number',
+                                              direction='Input',
+                                              datatype='GPFeatureRecordSetLayer',
+                                              parameterType='Required',
+                                              enabled=True,
+                                              multiValue=False)
+        input_layer_file_path = os.path.join(os.path.dirname(os.path.dirname(__file__)),
+                                             "layers",
+                                             "RelativeNumberFeaturesAreaInput.lyr")
+        input_area_features.value = input_layer_file_path
+
+        input_number_features = arcpy.Parameter(name='input_point_features',
+                                               displayName='Features to Number',
+                                               direction='Input',
+                                               datatype='GPFeatureLayer',
+                                               parameterType='Required',
+                                               enabled=True,
+                                               multiValue=False)
+
+        field_to_number = arcpy.Parameter(name='field_to_number',
+                                          displayName='Field to Number',
+                                          direction='Input',
+                                          datatype='Field',
+                                          parameterType='Optional',
+                                          enabled=True,
+                                          multiValue=False)
+        field_to_number.filter.list = ['Short', 'Long', 'Double', 'Single']
+        field_to_number.parameterDependencies = [input_number_features.name]
+
+        output_features= arcpy.Parameter(name='output_features',
+                                         displayName='Output Numbered Features',
+                                         direction='Output',
+                                         datatype='DEFeatureClass',
+                                         parameterType='Optional',
+                                         enabled=True,
+                                         multiValue=False)
+
+        layerFile = "NumberedStructures.lyr"
```

Comment:
@BobBooth - here is the Pro vs. ArcMap layer file check [we discussed yesterday](https://github.com/Esri/military-tools-geoprocessing-toolbox/issues/341#issuecomment-396998765)